recipe/meta.yaml

```
@@ -1,55 +1,186 @@
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 5dcb6b53bf6ce3ad8d9c881dc5325917ada258efb971f257786ef1c050668424
+  sha256: 575a9ef9444eda55a087c0201345d12ce1743f702538cf3170803a4c57d671e6
 
 build:
-  noarch: python
   number: 0
-  entry_points:
-    - lumen = lumen.command:main
-  script:
-    - {{ PYTHON }} -m pip install . -vv
-
-requirements:
-  host:
-    - python >=3.7
-    - pip
-    - param >=1.7.0
-    - pyct >=0.4.4
-    - setuptools >=30.3.0
-  run:
-    - python >=3.7
-    - panel >=0.11
-    - hvplot
-    - holoviews >=1.14
-    - pandas
-
-test:
-  imports:
-    - lumen
-  requires:
-    - pip
-  commands:
-    - pip check
-    - lumen --help
+
+
+outputs:
+  - name: {{ name|lower }}
+    build:
+      entry_points:
+        - lumen = lumen.command:main
+      skip: True  # [py<310]
+      script: python -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vv
+    requirements:
+      host:
+        - python
+        - pip
+        - hatchling
+        - hatch-vcs
+      run:
+        - python
+        - panel >=1.5.0
+        - hvplot
+        - holoviews >=1.17.0
+        - pandas
+        - bokeh
+        - intake <2
+        - jinja2 >3.0
+        - numpy
+        - packaging
+        - param >=1.9.0
+        - sqlglot
+        - pydantic-extra-types
+        - jsonschema
+        - nbformat
+      run_constrained:
+        - instructor >=1.6.4
+        - pydantic >=2.8.0
+        - panel-graphic-walker >=0.6.4
+        - llama-cpp-python >=0.3.0
+    test:
+      source_files:
+        - lumen/tests
+      imports:
+        - lumen
+        - lumen.ai
+        - lumen.auth
+        - lumen.command
+        - lumen.tests
+        - lumen.util
+        - lumen.validation
+        - lumen.views
+      requires:
+        - pip
+        - pydantic
+        - duckdb
+        - instructor
+        - griffe
+        - pydantic-extra-types
+        - jsonschema
+        - nbformat
+      commands:
+        - pip check
+        - lumen --help
+    about:
+      home: https://github.com/holoviz/lumen
+      summary: 'Illuminate your data. A monitoring solution built on Panel'
+      description: |
+        The Lumen project provides a framework for visual analytics, which allows users to
+        build data-driven dashboards from a simple yaml specification. The power of
+        Lumen comes from the ability to leverage the powerful data intake, data processing
+        and data visualization libraries available in the PyData ecosystem.
+      license: BSD-3-Clause
+      license_file: LICENSE
+      license_family: BSD
+      doc_url: https://lumen.holoviz.org/
+      dev_url: https://github.com/holoviz/lumen
+
+  - name: {{ name|lower }}-ai-local
+    build:
+      skip: True  # [py<310]
+    requirements:
+      host:
+        - python
+        - pip
+      run:
+        - python
+        - {{ pin_subpackage('lumen', exact=True) }}
```

```suggestion
        - {{ pin_subpackage('lumen-ai', exact=True) }}
```
It has to depend on output lumen-ai instead. Where lumen-ai is an output matching https://github.com/holoviz/lumen/blob/v0.9.0/pyproject.toml#L51

---

Added, this caused an all platform failure. I will take a deeper look into this

---

It seems that while building, the subpackage version info for 'lumen-ai' can not be found.  `ValueError: Didn't find subpackage version info for 'lumen-ai', which is used in a pin_subpackage expression.  Is it actually a subpackage?  If not, you want pin_compatible instead.`

---

Seems we need to add an older version of `Anthropic `to the list to build! looking into it now.

---

You need to add an output lumen-ai with deps matching https://github.com/holoviz/lumen/blob/v0.9.0/pyproject.toml#L51.

Then all the lumen-ai-* outputs depend on that lumen-ai output.